vignettes/page_plan.Rmd

```diff
@@ -0,0 +1,350 @@
+
+
+Examples of each of these approaches are below. 
+
+### Values-driven splitting
+
+Let's take a subset of our example demography data. 
+<details>
+
+<summary>Expand for the code used to produce this subset</summary>
+
+```{r}
+data_demog2 <- data_demog %>% 
+  filter(rowlbl1 %in% unique(rowlbl1)[1:3])
+```
+
+</details>
+
+```{r}
+head(data_demog2)
+```
+
+The formatted table as a single page is as follows:
+```{r}
+tfrmt(
+  # specify columns in the data
+  group = c(rowlbl1,grp),
+  label = rowlbl2,
+  column = column, 
+  param = param,
+  value = value,
+  sorting_cols = c(ord1, ord2),
+  # specify value formatting 
+  body_plan = body_plan(
+    frmt_structure(group_val = ".default", label_val = ".default", frmt_combine("{n} {pct}", 
+                                                                                n = frmt("xxx"),
+                                                                                pct = frmt_when("==100" ~ "",
+                                                                                                "==0" ~ "",
+                                                                                                TRUE ~ frmt("(xx.x %)")))),
+    frmt_structure(group_val = ".default", label_val = "n", frmt("xxx")),
+    frmt_structure(group_val = ".default", label_val = c("Mean", "Median", "Min","Max"), frmt("xxx.x")),
+    frmt_structure(group_val = ".default", label_val = "SD", frmt("xxx.xx")),
+    frmt_structure(group_val = ".default", label_val = ".default", p = frmt("")),
+    frmt_structure(group_val = ".default", label_val = c("n","<65 yrs","<12 months","<25"), p = frmt_when(">0.99" ~ ">0.99",
+                                                                                                          "<0.001" ~ "<0.001",
+                                                                                                          TRUE ~ frmt("x.xxx", missing = "")))
+  ),
+  # remove extra cols
+  col_plan = col_plan(-grp, 
+                      -starts_with("ord") ),
+  # Specify column styling plan
+  col_style_plan = col_style_plan(
+    col_style_structure(align = c(".",","," "), col = vars(Placebo, contains("Dose"), "Total", "p-value"))
+  ),
+  # Specify row group plan
+  row_grp_plan = row_grp_plan(
+    row_grp_structure(group_val = ".default", element_block(post_space = " ")),
+    label_loc = element_row_grp_loc(location = "column")
+  )
+)  %>% 
+  print_to_gt(data_demog2) 
+```
+
+#### Every unique value of 1 variable
+
+Suppose we want to split the table by every unique value of `rowlbl1`. and add a footnote indicating the table grouping. We can drop `rowlbl1` from the tables since its value will be printed in the note. We will use the `gt::grp_pull()` function to print the individual tables nicely in the vignette. 
+```{r  }
+gts <- tfrmt(
```

Comment:
is there an easy way to make this shorter and then reference the vignette where it shows how to make this in full. Then you can just focus on the page plan (like we might want to save the pre-made ones as json files in the inst folder and then we can just reference them and say the styling for this tfrmt can be found in this vignette)

Reply:
Yes! thanks for the suggestion. As discussed, going to make use of `layer_tfrmt` here.